docs/developers/content-creation/creating-a-role.md

```diff
@@ -40,3 +44,526 @@ suicideMultiplier = -1
 ```
 
 [[Find them in the role base]](https://github.com/TTT-2/TTT2/blob/master/lua/terrortown/entities/roles/ttt_role_base/shared.lua)
+
+## Setting up the files 
+
+Before we can dive into the lua programming, we must set up the role’s file structure, so that the role is detected by TTT2 as such. But this is not only depending on the file structure. Another important aspect is the initial code, which is the same for every role.
+
+### Folder structure
+
+Every role has the same folder structure. As a little side note: You can name the role’s folder like you want, but if you want to fit the most other roles, you should name the folder  `ttt2-role_[abbreviation of the role]”`.
+
+Inside this folder you will need another folder that must be called “lua” and then “terrortown”. In the terrortown-folder the structure is split into 3 parts:
```

Comment:
you should add maybe somthing like I did here to show the folder structure. It is hard to understand yb just reading: https://docs.ttt2.neoxult.de/developers/content-creation/creating-a-hud-theme/